infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironment.java

```diff
@@ -63,13 +66,24 @@ public OpenShiftEnvironment(KubernetesEnvironment k8sEnv) {
   public OpenShiftEnvironment(
       InternalEnvironment internalEnvironment,
       Map<String, Pod> pods,
+      Map<String, Deployment> deployments,
+      Map<String, PodData> podData,
```

Comment:
I think it's not a good idea to have the ability to have a public constructor where `deployments`, `pods` and `podData` are set by different arguments. As far as I see there are no usages of this constructor but still, Environment created via this constructor may be in inconsistency state.
We are able to avoid even having such protected constructor by moving `podsData` filling logic to `KubernetesEnvironment` constructor from Builders. WDYT?

Reply:
I think that's a great idea. I'll do that.